Sources/PackageModel/Target.swift

```diff
@@ -443,9 +443,80 @@ public final class ClangTarget: Target {
 }
 
 public final class BinaryTarget: Target {
+    /// The kind of binary artifact.
+    public let kind: Kind
 
     /// The original source of the binary artifact.
-    public enum ArtifactSource: Equatable, Codable {
+    public let origin: Origin
+
+    /// The binary artifact path.
+    public var artifactPath: AbsolutePath {
+        return self.sources.root
+    }
+
+    public init(
+        name: String,
+        kind: Kind,
+        platforms: [SupportedPlatform] = [],
+        path: AbsolutePath,
+        origin: Origin
+    ) {
+        self.origin = origin
+        self.kind = kind
+        let sources = Sources(paths: [], root: path)
+        super.init(
+            name: name,
+            defaultLocalization: nil,
+            platforms: platforms,
+            type: .binary,
+            sources: sources,
+            dependencies: [],
+            buildSettings: .init()
+        )
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case kind
+        case origin
+        case artifactSource // backwards compatibility 2/2021
+    }
+
+    public override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(self.origin, forKey: .origin)
+        try container.encode(self.kind, forKey: .kind)
+    }
+
+    // 👀 do we need backwards compatibility?
```

Comment:
not sure when this gets encoded / decoded and the requirements around backwards compatibility for that

Reply:
I would assume that this is only for caching.  But I don't have definite evidence.  Certainly it's not a versioned format of any kind.

Reply:
And this is not actually the transport protocol for the structure coming from the package description.